x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2338,4 +2338,12 @@ public void testInInRangeValues() {
         PhysicalPlan p = testContext.optimizeAndPlan("SELECT long FROM test WHERE long IN (1, 2, 3, " + Long.MAX_VALUE + ", 5, 6, 7)");
         assertEquals(EsQueryExec.class, p.getClass());
     }
+    
+    public void testCombineBinaryComparisonsBugKeyword() {
```

Comment:
I'd add a simple numeric test in x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRulesTests.java to replicate and check the fix, as well as more accurately inspect the translation.

In that respect, not sure if this particular test would still be needed, but if you decide to keep it, I'd also add a comment on the condition that the value of `negatedText` fulfils as it's not obvious.
Also, the "Bug" in the title is not explanatory: either rename the test or add a reference.

Reply:
Thanks for the comment. Cleaned up the draft, added additional test cases that were discovered by randomized tests (will add randomized tests in separate PR), added an integration test for this particular bug.